### PR TITLE
Remove 'includes' API, fix CI build failure

### PR DESCRIPTION
**Pull request by Alanscut**
### Changes:

1. Error message: "Object doesn't support property or method 'includes' to equal."

2. Reason: 'includes' belongs to the API of `ES7`, `ie11` does not support it.

    https://github.com/axios/axios/blob/master/lib/utils.js#L29

3. Related PR #1816

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -26,7 +26,8 @@ function isArray(val) {
  * @returns {boolean} True if value is a Buffer, otherwise false
  */
 function isBuffer(val) {
-  return ![undefined, null].includes(val) && val.constructor === Buffer;
+  if (val === null || val === void 0 || val.constructor === null || val.constructor === void 0 )  return false;
```
Comment (yasuf) [inline lib/utils.js]: what's the difference between using `void 0` and undefined?

Comment (Alanscut) [reply lib/utils.js]: `undefined` is not a reserved word in javascript, so it can be rewritten. `void 0` is stricter.

Comment (yasuf) [reply lib/utils.js]: we rely on `type of x === 'undefined'` throughout the file, might be better to be consistent, also are there cases where the constructor can be null? could it be sufficient to just check `!isUndefined(val.constructor)`, also we can just use our own helper? `isUndefined`

Comment (Alanscut) [reply lib/utils.js]: I agree to use `isUndefined`, and I suggest to replace all the other `undefined` judgement with our own helper function `isUndefined`, if you agree, then I will submit a PR.

if `constructor` has been overwriten with `null` (extreme paticular case), then `typeof val.constructor.isBuffer === 'function'` will report an error.

**Comment by Alanscut** [comment]
> @Alanscut Can we simply copy the implementation from [is-buffer](https://github.com/feross/is-buffer/blob/master/index.js), as well as [its tests](https://github.com/feross/is-buffer/blob/master/test/basic.js)? Of course, with a reference comment. That looks clear and simple. And don't change the position of function `isUndefined` if not necessary.

@chinesedfan because we have to obey the  3 rules of eslint:
 * "no-use-before-define": 2,       // http://eslint.org/docs/rules/no-use-before-define
*  "eqeqeq": 2,                     // http://eslint.org/docs/rules/eqeqeq
* "no-eq-null": 2,                 // http://eslint.org/docs/rules/no-eq-null

File: lib/utils.js [hunk 2]
```diff
@@ -26,7 +35,8 @@ function isArray(val) {
  * @returns {boolean} True if value is a Buffer, otherwise false
  */
 function isBuffer(val) {
-  return ![undefined, null].includes(val) && val.constructor === Buffer;
+  if (val === null || isUndefined(val) || val.constructor === null ||  isUndefined(val.constructor))  return false;
+  return typeof val.constructor.isBuffer === 'function' && val.constructor.isBuffer(val);
```
Comment (chinesedfan) [inline lib/utils.js]: @Alanscut Thanks for reminding me lint rules. Then can we write all conditions in the same statement? And you have an extra space before `isUndefined` and `return false` now. :P

```
return val !== null && !isUndefined(val) && val.constructor !== null && !isUndefined(val.constructor)
    && typeof val.constructor.isBuffer === 'function' && val.constructor.isBuffer(val);
```

Comment (Alanscut) [reply lib/utils.js]: agree, this will be more clear!

**Pull request merged**